src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -299,21 +299,22 @@ void primitive_inst::update_impl() {
         };
 
         auto layout_key = get_layout_key();
-        auto& cache = _network.get_program()->get_implementations_cache();
+        auto& cache = get_network().get_program()->get_implementations_cache();
         if (cache.has(layout_key)) {
             _impl = cache.get(layout_key)->clone();
             GPU_DEBUG_PROFILED_STAGE_CACHE_HIT(true);
         } else {
             auto lru = cache.get_lru_element();
             _impl = _node.type()->choose_impl(_node, *_impl_params);
+            _impl->update_kernels(get_network().get_kernels_cache());
+            get_network().compile();
+            _impl->init_kernels(get_network().get_kernels_cache());
             bool lru_popped = cache.add(layout_key, _impl->clone());
             if (lru_popped) {
                 for (auto& id : lru->get_kernel_ids())
-                    _network.get_program()->remove_kernel(id);
+                    get_network().remove_kernel(id);
```

Comment:
I am afraid this might cause the memory leak, becuase it only removes the kernels cache in its own network, though the other network might have this kernel too...

Reply:
Done